service/src/test/java/com/ge/predix/controller/test/PolicyEvalWithGraphDbControllerIT.java

```diff
@@ -202,6 +276,30 @@ public void testPolicyEvaluation(final Zone zone, final PolicySet testPolicySet,
                 Effect.PERMIT };
     }
 
+    /**
+     * Test that evaluation is successful when the resource and/or subject attributes exceed the length. The policy set will return
+     * indeterminate because the traversal limit is exceeded.
+     */
+    Object[] evaluationWithResourceAttributesExceedingTraversalLimitData() {
+        String errorMessage = "The number of attributes on this resource "
+                + EVIDENCE_SCULLYS_TESTIMONY_ID + " has exceeded the maximum limit of " + TEST_TRAVERSAL_LIMIT;
+        return new Object[] { this.testZone1, this.policySet, createThreeLevelResourceHierarchy(), null,
+                createPolicyEvalRequest("GET", EVIDENCE_SCULLYS_TESTIMONY_ID, AGENT_MULDER),
+                Effect.INDETERMINATE, errorMessage};
+    }
+
+    /**
+     * Test that subjects and resources inherit attributes from their parents. The policy set will permit the request
+     * if the subject and resource successfully inherit the required attributes from their respective parents.
+     */
+    Object[] evaluationWithSubjectAttributesExceedingTraversalLimitData() {
+        String errorMessage = "The number of attributes on this subject "
```

Comment:
"The number of attributes on subject '\<subject id\>' exceeds the maximum of \<limit\>."